src/components/Tool/FixEmailAddresses/FixEmailAddressPerson.tsx

```diff
@@ -132,7 +132,7 @@ export const FixEmailAddressPerson: React.FC<FixEmailAddressPersonProps> = ({
         personId: personId,
         isPrimary: email.primary,
       })) || [],
-    [emailAddresses],
+    [emailAddresses, dataState],
```

Comment:
Why do you need `dataState` as a dependency?

Reply:
In my testing, `emailAddresses` did not update when changing data on it and `useMemo` never got called, but it did get called when `dataState` changed.